omniisaacgymenvs/robots/sensors/exteroceptive/camera_module_generator.py

```diff
@@ -0,0 +1,216 @@
+        UsdGeom.Xformable(prim).AddTranslateOp()
+        UsdGeom.Xformable(prim).AddRotateXYZOp()
+        prim.GetAttribute('xformOp:translate').Set((0, 0, 0))
+        prim.GetAttribute('xformOp:rotateXYZ').Set((0, 0, 0))
+        UsdPhysics.RigidBodyAPI.Apply(prim)
+
+        camera_body = add_reference_to_stage(sensor_body_usd, os.path.join(self.root_prim_path, sensor_link, "base_body"))
+        UsdGeom.Xformable(camera_body).AddTranslateOp()
+        UsdGeom.Xformable(camera_body).AddRotateXYZOp()
+        camera_body.GetAttribute('xformOp:translate').Set((0, 0, 0))
+        camera_body.GetAttribute('xformOp:rotateXYZ').Set((0, 0, 0))
+        UsdPhysics.CollisionAPI.Apply(camera_body)
+    
+    def _add_link(self, link_name:str) -> None:
+        """
+        Add link(body).
+        Args:
+            link_name (str): name of the link."""
+        
+        self.stage.DefinePrim(os.path.join(self.root_prim_path, link_name), "Xform")
+
+    def _add_transform(self, link_name:str, transform:list) -> None:
+        """
+        Add transform to the link(body) relative to its parent prim.
+        Args:
+            link_name (str): name of the link.
+            transform (list): transform of the link."""
+        
+        prim = get_prim_at_path(os.path.join(self.root_prim_path, link_name))
+        UsdGeom.Xformable(prim).AddTranslateOp()
+        UsdGeom.Xformable(prim).AddRotateXYZOp()
+        prim.GetAttribute('xformOp:translate').Set(Gf.Vec3f(*transform[:3]))
+        prim.GetAttribute('xformOp:rotateXYZ').Set(Gf.Vec3f(*transform[3:]))
+        UsdPhysics.RigidBodyAPI.Apply(prim)
+
+    def _add_joint(self, parent_link_name:str, child_link_name:str, joint_name:str, transform:list=[0, 0, 0, 0, 0, 0]) -> None:
+        """
+        Add joint between two links(bodies).
+        Args:
+            parent_link_name (str): name of the parent link.
+            child_link_name (str): name of the child link.
+            joint_name (str): name of the joint.
+            transform (list): transform of the joint."""
+        
+        translate = Gf.Vec3f(*transform[:3])
+        rotatation_quat = euler_angles_to_quat(euler_angles=transform[3:], degrees=True)
+        rotation = Gf.Quatf(rotatation_quat[0], *rotatation_quat[1:].tolist())
+
+        # define a joint
+        fixedJoint = UsdPhysics.FixedJoint.Define(self.stage, os.path.join(self.root_prim_path, self.sensor_base["prim_name"], joint_name))
+        fixedJoint.CreateBody0Rel().SetTargets( [Sdf.Path(os.path.join(self.root_prim_path, parent_link_name))])
+        fixedJoint.CreateBody1Rel().SetTargets( [Sdf.Path(os.path.join(self.root_prim_path, child_link_name))])
+
+        fixedJoint.CreateLocalPos0Attr().Set(translate)
+        fixedJoint.CreateLocalRot0Attr().Set(rotation)
+        fixedJoint.CreateLocalPos1Attr().Set(Gf.Vec3f(0.0, 0.0, 0.0))
+        fixedJoint.CreateLocalRot1Attr().Set(Gf.Quatf(1.0))
```

Comment:
These methods, or some very similar methods are given in robots / mfp_utils.py or something like that. I'm a bit torn as to the camera should be added by the RL script or be coming with the robot's USD. And you would just need to find it in the stage. I think having an option to add a camera that's not here is nice, but is there an option to use an existing camera? Most people will use pre-configured robots.

Reply:
here are two things I did. 

1. Create a RealSense prim structure at the same time as floating platform is generated
2. Made sensor parameter overridable in camera.py (which is mainly replicator code)